In MySQL 5.5, the InnoDB Plugin is the default version and it's compiled in.
In this case, the plugin cannot be uninstalled and we get error when trying
to do so:

mysql> uninstall plugin innodb;
ERROR 1305 (42000): PLUGIN innodb does not exist

But the error message is misleading. The plugin does exist, it just cannot
be installed because it's not dynamically loaded.

MySQL bug report: http://bugs.mysql.com/bug.php?id=51771
MariaDB bug report: https://mariadb.atlassian.net/browse/MDEV-4573
MariaDB fix: http://bazaar.launchpad.net/~maria-captains/maria/5.5/revision/3802

diff -up mysql-5.5.34/include/mysql.h.p28 mysql-5.5.34/include/mysql.h
--- mysql-5.5.34/include/mysql.h.p28	2013-09-09 20:01:38.000000000 +0200
+++ mysql-5.5.34/include/mysql.h	2013-10-10 12:09:03.934687127 +0200
@@ -134,6 +134,7 @@ typedef unsigned long long my_ulonglong;
 
 /* backward compatibility define - to be removed eventually */
 #define ER_WARN_DATA_TRUNCATED WARN_DATA_TRUNCATED
+#define WARN_PLUGIN_DELETE_BUILTIN ER_PLUGIN_DELETE_BUILTIN
 
 typedef struct st_mysql_rows {
   struct st_mysql_rows *next;		/* list of rows */
diff -up mysql-5.5.34/sql/share/errmsg-utf8.txt.p28 mysql-5.5.34/sql/share/errmsg-utf8.txt
--- mysql-5.5.34/sql/share/errmsg-utf8.txt.p28	2013-09-09 20:01:38.000000000 +0200
+++ mysql-5.5.34/sql/share/errmsg-utf8.txt	2013-10-10 12:09:03.936687128 +0200
@@ -6161,7 +6161,7 @@ WARN_NO_MASTER_INFO
 WARN_OPTION_IGNORED
   eng "<%-.64s> option ignored"
   ger "Option <%-.64s> ignoriert"
-WARN_PLUGIN_DELETE_BUILTIN
+ER_PLUGIN_DELETE_BUILTIN
   eng "Built-in plugins cannot be deleted"
   ger "Eingebaute Plugins können nicht gelöscht werden"
 WARN_PLUGIN_BUSY
diff -up mysql-5.5.34/sql/sql_plugin.cc.p28 mysql-5.5.34/sql/sql_plugin.cc
--- mysql-5.5.34/sql/sql_plugin.cc.p28	2013-10-10 12:09:03.837687101 +0200
+++ mysql-5.5.34/sql/sql_plugin.cc	2013-10-10 12:09:03.937687128 +0200
@@ -1929,9 +1929,7 @@ bool mysql_uninstall_plugin(THD *thd, co
   }
   if (!plugin->plugin_dl)
   {
-    push_warning(thd, MYSQL_ERROR::WARN_LEVEL_WARN,
-                 WARN_PLUGIN_DELETE_BUILTIN, ER(WARN_PLUGIN_DELETE_BUILTIN));
-    my_error(ER_SP_DOES_NOT_EXIST, MYF(0), "PLUGIN", name->str);
+    my_error(ER_PLUGIN_DELETE_BUILTIN, MYF(0));
     goto err;
   }
   if (plugin->load_option == PLUGIN_FORCE_PLUS_PERMANENT)
